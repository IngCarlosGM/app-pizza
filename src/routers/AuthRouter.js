import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';

import { Login } from '../components/auth/Login';

export const AuthRouter = () => {
    return (
        <main className="auth__main">
            <Switch>
                <Route 
                    exact
                    path="/auth/login"
                    component={ Login }
                />

                <Redirect to="/auth/login" />


            </Switch>
        </main>
    )
}
