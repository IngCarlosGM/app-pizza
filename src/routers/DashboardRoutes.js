import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { Dashboard } from '../components/dashboard/Dashboard';

export const DashboardRoutes = ({ location }) => {

    localStorage.setItem('lastPath', location.pathname+location.search);

    return (
        <>
            <Switch>
                <Route path="/" component={ Dashboard }/>
                <Route exact path="/search/:q" component={ Dashboard }/>
                <Route exact path="/search/:id/:show" component={ Dashboard }/>
                <Route exact path="/search/:q/:id/:show" component={ Dashboard }/>

                <Redirect to="/" />
            </Switch>
        </>
    )
}

