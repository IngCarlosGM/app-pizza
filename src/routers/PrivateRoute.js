import React from 'react';
import PropTypes from 'prop-types';

import { Route, Redirect } from 'react-router-dom';


export const PrivateRoute = ({
    loggedIn,
    component: Component,
    ...rest
}) => {
    
    return (
        <Route { ...rest }
            component={ (props) => (
                ( loggedIn )
                    ? ( <Component { ...props } /> )
                    : ( <Redirect to="/auth/login" /> )
            )}
        
        />
    )
}

PrivateRoute.propTypes = {
    loggedIn: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}
