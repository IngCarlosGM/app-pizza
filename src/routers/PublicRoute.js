import React from 'react';
import PropTypes from 'prop-types';

import { Route, Redirect } from 'react-router-dom';


export const PublicRoute = ({
    loggedIn,
    component: Component,
    ...rest
}) => {

    return (
        <Route { ...rest }
            component={ (props) => (
                ( loggedIn )
                    ? ( <Redirect to="/" /> )
                    : ( <Component { ...props } /> )
            )}
        
        />
    )
}

PublicRoute.propTypes = {
    loggedIn: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}
