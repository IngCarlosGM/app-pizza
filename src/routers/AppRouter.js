import React, { useEffect, useState } from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Redirect
  } from 'react-router-dom';

import { AuthRouter } from './AuthRouter';
import { Dashboard } from '../components/dashboard/Dashboard';
import { useDispatch } from 'react-redux';
// import { login } from '../actions/auth';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';
import BackgroundAside from '../components/BackgroundAside';

export const AppRouter = () => {

    const dispatch = useDispatch();

    const [checking, setChecking] = useState(true);
    const [loggedIn, setLoggedIn] = useState(false);

    useEffect(() => {
        if( !loggedIn ) {
            // setLoggedIn( true );
            setChecking( false );
        }
        
    }, [ dispatch, setChecking, setLoggedIn, loggedIn ]);

    if( checking ) {
        return (
            <h1>Wait...</h1>
        );
    }

    return (
        <Router>
            <div className="grid-layaout overflow-hidden">
                
                <BackgroundAside />
                
                <Switch>


                    <PublicRoute
                        path="/auth"
                        component={ AuthRouter }
                        loggedIn= { loggedIn }
                    />

                    <PrivateRoute 
                        exact
                        path="/"
                        component={ Dashboard }
                        loggedIn={ loggedIn }
                    />

                    <Redirect to="/auth/login" />

                </Switch>
            </div>
        </Router>
    )
}
