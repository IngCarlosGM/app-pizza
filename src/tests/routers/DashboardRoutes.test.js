import React from 'react';
import { mount } from 'enzyme';
import { DashboardRoutes } from '../../routers/DashboardRoutes';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
    auth: {
        name: 'Pepito Perez',
        logged: true
    },
    ui: {
        loading: false,
        msgError: null
    },
    dashboard: {
        
    }
}

let store = mockStore(initialState);


describe('Pruebas en <DashboardRoutes />', () => {

    const props = {
        location: {
            pathname: '/',
            search: ''
        }
    }

    test('should render correctly', () => {
        
        const wrapper = mount(
            <Provider store={ store } >
                <MemoryRouter>
                    <DashboardRoutes {...props}/>
                </MemoryRouter>
            </Provider>
        );

        expect( wrapper ).toMatchSnapshot();
    })
    
})
