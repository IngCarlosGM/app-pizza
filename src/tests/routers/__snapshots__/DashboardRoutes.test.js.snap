// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes /> should render correctly 1`] = `
<div
  className="scroll-y container-dashboard"
>
  <nav
    className="align-center-right"
  >
    <img
      alt="ícono candado"
      className="img_responsive size-icon-logout"
      src="../assets/ic_contrasena.png"
    />
    <span
      className="text-logout pointer"
      onClick={[Function]}
    >
      Salir
    </span>
  </nav>
  <span
    className="route-navigation"
  >
    Pizzerías
  </span>
  <h1
    className=" mt-secondary"
  >
    Tiendas
  </h1>
  <div
    className="description-page "
  >
    Escoge tu pizzería favorita
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input-search input--block mt-third"
      name="searchText"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="Buscar una tienda"
      type="text"
      value=""
    />
    <button
      className="btn__primary btn-block btn-responsive"
      id="btn_search"
      type="submit"
    >
      Buscar
    </button>
  </form>
  <div
    className="mt-third not-store-founds"
  >
    No se han encontrado tiendas que coincidan con la búsqueda
  </div>
  <div
    className="grid-shop mt-secondary"
  />
  <footer
    className="space-between align-center"
  >
    <div>
      <a
        href="https://facebook.com"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Logo best pizza blanco"
          className="footer__icons img-responsive pointer"
          src="../assets/logo-facebook.png"
        />
      </a>
      <a
        href="https://instagram.com"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Logo best pizza blanco"
          className="footer__icons img-responsive pointer"
          src="../assets/logo-instagram.png"
        />
      </a>
    </div>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Logo best pizza blanco"
        className="footer__icons img-responsive pointer"
        src="../assets/logo-best-pizza-white.png"
      />
    </a>
  </footer>
</div>
`;
